chromadb/server/fastapi/__init__.py

```diff
@@ -77,13 +78,24 @@ def use_route_names_as_operation_ids(app: _FastAPI) -> None:
 async def catch_exceptions_middleware(
     request: Request, call_next: Callable[[Request], Any]
 ) -> Response:
-    try:
-        return await call_next(request)
-    except ChromaError as e:
-        return fastapi_json_response(e)
-    except Exception as e:
-        logger.exception(e)
-        return JSONResponse(content={"error": repr(e)}, status_code=500)
+    with span_proxy_context("FastAPI.catch_exceptions_middleware") as _:
+        try:
+            return await call_next(request)
+        except Exception as e:
+            trace_id = OpenTelemetryClient.get_current_span_id()
```

Comment:
What happens here if we're not in a span?

Reply:
We pass None to headers and exception, which results in the header or body attribute in an error response not being sent back to the client.